### PR TITLE
feat: adds tests for TokenVesting

### PR DESCRIPTION
**Pull request by Aniket965**

### CODE REVIEW

File: test/tokenVesting.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
+import { BigNumber, Signer } from "ethers";
+import { TokenVesting, UnrealToken } from "../typechain";
+import chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+const { ethers } = require("hardhat");
```
Comment (unrealfi) [inline test/tokenVesting.ts]: Lets use imports here as well?

File: test/tokenVesting.ts [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  return ethers.provider.send("evm_increaseTime", [seconds]);
+}
+
+async function mineOneBlock() {
+  return ethers.provider.send("evm_mine");
+}
+
+function getUnixTimeStamp(date: Date) {
+  return (date.getTime() / 1000).toFixed(0);
+}
```
Comment (unrealfi) [inline test/tokenVesting.ts]: It would be great if we can move these functions to `utils`

File: test/tokenVesting.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+const avg_seconds_in_months = 60 * 60 * 24 * 30.8;
+describe("Token Vesting", function () {
+  const vestedSupply = ethers.utils.parseEther("1000000000");
+  const vestingSupply = ethers.utils.parseEther("5000000000");
```
Comment (unrealfi) [inline test/tokenVesting.ts]: rename please

File: test/tokenVesting.ts [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    endTime.setMonth(startTime.getMonth() + 3);
+
+    const startTimeStamp = getUnixTimeStamp(startTime);
+    const endTimeStamp = getUnixTimeStamp(endTime);
+
+    const account1 = await accounts[3].getAddress();
+    const account2 = await accounts[4].getAddress();
+    const dummyCommunity = [account1, account2];
+    const dummyAmounts = [amount1, amount2];
+    const immidiateRelease = BigNumber.from("0");
```
Comment (unrealfi) [inline test/tokenVesting.ts]: typo

File: test/tokenVesting.ts [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    endTime.setMonth(startTime.getMonth() + 3);
+
+    const startTimeStamp = getUnixTimeStamp(startTime);
+    const endTimeStamp = getUnixTimeStamp(endTime);
+
+    const account1 = await accounts[3].getAddress();
+    const account2 = await accounts[4].getAddress();
+    const dummyCommunity = [account1, account2];
+    const dummyAmounts = [amount1, amount2];
+    const immidiateRelease = BigNumber.from("10");
```
Comment (unrealfi) [inline test/tokenVesting.ts]: maybe we should declare immediateRelease as a global variable and mutate it within testcases

**Comment by Aniket965** [comment]
@unrealfi  fixed suggestions.

**Review by unrealfi — APPROVED** [review]
Merging!

**Pull request merged**